pkg/parse/root.go

```diff
@@ -261,6 +261,18 @@ func setSourceStatusFields(source *v1beta1.SourceStatus, newStatus *SourceStatus
 	source.LastUpdate = newStatus.LastUpdate
 }
 
+func (p *root) setPreSyncAnnotations(ctx context.Context, commit string) status.Error {
+	rs := &v1beta1.RootSync{}
+	if err := p.Client.Get(ctx, rootsync.ObjectKey(p.SyncName), rs); err != nil {
+		return status.APIServerError(err, "failed to get RootSync for parser")
+	}
+	existing := rs.DeepCopy()
+	core.SetAnnotation(rs, metadata.SourceCommitAnnotationKey, commit)
```

Comment:
Suggest checking the existing annotations to avoid unnecessary updates if they're the same.

Reply:
Added check. Also added similar check in resetSourceAnnotations when removing the annotations when sourceType switched or set to Git.